### ISSUE TITLE
Migration - Mysql - on delete, on update issue

### ISSUE THREAD

**Issue by Michal-St** [phalcon/phalcon-devtools#925]
I have issue with migration. I'm using Mysql DB and after migration to another server  DevTools doesn't remember type of relationship on foregin keys. This case is very often in intersactions tables. For example when I set "On delete: Cascade" after migration I have "On delete: Restricted"

Do you have the same problem with that?

**Comment by sergeyklay** [comment]
### Expected and Actual Behavior

> **Describe what you are trying to achieve and what goes wrong.**

> Provide output if related

```php
// paste output here
```
> Provide minimal script to reproduce the issue

```php
// paste code
```

### Details

* System info and versions (if possible): (`phalcon info`)
* Phalcon Framework version: (`php --ri phalcon`)
* PHP Version: (`php -v`)
* Operating System:
* Server: Nginx | Apache | Other
* Other related info (Database, table schema):

**Comment by Michal-St** [comment]
SQL (MysqlDB)

> -- ----------------------------
> -- Table structure for tag
> -- ----------------------------
> DROP TABLE IF EXISTS `tag`;
> CREATE TABLE `tag` (
>   `tag_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
>   `name` varchar(255) NOT NULL,
>   PRIMARY KEY (`tag_id`),
>   UNIQUE KEY `tag_nam_index` (`name`) USING BTREE
> ) ENGINE=InnoDB AUTO_INCREMENT=358 DEFAULT CHARSET=utf8;
> 
> -- ----------------------------
> -- Table structure for tag_group
> -- ----------------------------
> DROP TABLE IF EXISTS `tag_group`;
> CREATE TABLE `tag_group` (
>   `group_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
>   `name` varchar(255) NOT NULL,
>   `type` varchar(255) DEFAULT NULL,
>   PRIMARY KEY (`group_id`),
>   UNIQUE KEY `tag_group_name_unique` (`name`)
> ) ENGINE=InnoDB AUTO_INCREMENT=1207 DEFAULT CHARSET=utf8;
> 
> -- ----------------------------
> -- Table structure for tag_to_group
> -- ----------------------------
> DROP TABLE IF EXISTS `tag_to_group`;
> CREATE TABLE `tag_to_group` (
>   `tag_id` int(10) unsigned NOT NULL,
>   `group_id` int(10) unsigned NOT NULL,
>   PRIMARY KEY (`tag_id`,`group_id`),
>   KEY `tag_to_group_group_id_fk` (`group_id`),
>   CONSTRAINT `tag_to_group_group_id_fk` FOREIGN KEY (`group_id`) REFERENCES `tag_group` (`group_id`) ON DELETE CASCADE ON UPDATE CASCADE,
>   CONSTRAINT `tag_to_group_tag_id_fk` FOREIGN KEY (`tag_id`) REFERENCES `tag` (`tag_id`) ON DELETE CASCADE ON UPDATE CASCADE
> ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
> 


Migration files:

`<?php 

use Phalcon\Db\Column;
use Phalcon\Db\Index;
use Phalcon\Db\Reference;
use Phalcon\Mvc\Model\Migration;

/**
 * Class TagToGroupMigration_1013
 */
class TagToGroupMigration_1013 extends Migration
{
    /**
     * Define the table structure
     *
     * @return void
     */
    public function morph()
    {
        $this->morphTable('tag_to_group', array(
                'columns' => array(
                    new Column(
                        'tag_id',
                        array(
                            'type' => Column::TYPE_INTEGER,
                            'unsigned' => true,
                            'notNull' => true,
                            'size' => 10,
                            'first' => true
                        )
                    ),
                    new Column(
                        'group_id',
                        array(
                            'type' => Column::TYPE_INTEGER,
                            'unsigned' => true,
                            'notNull' => true,
                            'size' => 10,
                            'after' => 'tag_id'
                        )
                    )
                ),
                'indexes' => array(
                    new Index('PRIMARY', array('tag_id', 'group_id'), null),
                    new Index('tag_to_group_group_id_fk', array('group_id'), null)
                ),
                'references' => array(
                    new Reference(
                        'tag_to_group_group_id_fk',
                        array(
                            'referencedSchema' => 'tradevis',
                            'referencedTable' => 'tag_group',
                            'columns' => array('group_id'),
                            'referencedColumns' => array('group_id'),
                            'onUpdate' => '',
                            'onDelete' => ''
                        )
                    ),
                    new Reference(
                        'tag_to_group_tag_id_fk',
                        array(
                            'referencedSchema' => 'tradevis',
                            'referencedTable' => 'tag',
                            'columns' => array('tag_id'),
                            'referencedColumns' => array('tag_id'),
                            'onUpdate' => '',
                            'onDelete' => ''
                        )
                    )
                ),
                'options' => array(
                    'TABLE_TYPE' => 'BASE TABLE',
                    'AUTO_INCREMENT' => '',
                    'ENGINE' => 'InnoDB',
                    'TABLE_COLLATION' => 'utf8_general_ci'
                ),
            )
        );
    }

    /**
     * Run the migrations
     *
     * @return void
     */
    public function up()
    {

    }

    /**
     * Reverse the migrations
     *
     * @return void
     */
    public function down()
    {

    }

}
`



Details:

Phalcon DevTools (2.0.9)


php --ri phalcon

`Web framework delivered as a C-extension for PHP
phalcon => enabled
Author => Phalcon Team and contributors
Version => 2.0.10
Build Date => Apr 13 2016 10:22:07
Powered by Zephir => Version 0.9.2a-dev

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.db.force_casting => Off => Off
phalcon.orm.events => On => On
phalcon.orm.virtual_foreign_keys => On => On
phalcon.orm.column_renaming => On => On
phalcon.orm.not_null_validations => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.enable_literals => On => On
phalcon.orm.late_state_binding => Off => Off
phalcon.orm.enable_implicit_joins => On => On
phalcon.orm.cast_on_hydrate => Off => Off
phalcon.orm.ignore_unknown_columns => Off => Off


php -v 

```
PHP 5.6.16 (cli) (built: Nov 26 2015 07:52:51)
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies
    with Xdebug v2.3.3, Copyright (c) 2002-2015, by Derick Rethans
    with blackfire v1.9.0, https://blackfire.io, by Blackfireio Inc.

```
`


Centos, nginx, MySql

**Comment by Michal-St** [comment]
Tho most important is this code: 

`CONSTRAINT tag_to_group_group_id_fk FOREIGN KEY (group_id) REFERENCES tag_group (group_id) ON DELETE CASCADE ON UPDATE CASCADE,
CONSTRAINT tag_to_group_tag_id_fk FOREIGN KEY (tag_id) REFERENCES tag (tag_id) ON DELETE CASCADE ON UPDATE CASCADE`

and this is migration:

> `'onUpdate' => '',
'onDelete' => ''`

Still I'm using phalcon 2.* maybe this problem not exists in phalcon 3 and newest verson of devTools

**Comment by Jeckerson** [comment]
What is your version of MySQL? 5.5?

**Comment by Michal-St** [comment]
5.6.29

**Comment by sergeysviridenko** [comment]
It was fixed in devtools #1033 and in cphalcon https://github.com/sergeysviridenko/cphalcon/commit/9d6dfaad52db16cf57980a02bdae23b9fca8d895.
I guess this issue can be closed.
